test/integration/targets/aws_codebuild/defaults/main.yml

```diff
@@ -1,2 +1,7 @@
 ---
 # defaults file for aws_codebuild
+
+# IAM role names have to be less than 64 characters
+# The 8 digit identifier at the end of resource_prefix helps determine during
+# which test something was created and allows tests to be run in parallel
+iam_role_name: "ansible-test-sts-{{ resource_prefix | regex_search('([0-9]+)$') }}-codebuild-service-role"
```

Comment:
1. Why use sts in the role name?  This feels like we're just abusing IAM permissions that were granted for something else.
2. splitting off the ID at the end feels like something which should be refactored for the other places IAM roles are used.

All in, might it be worth ansible-test offering some form of "short_resource_prefix" which doesn't include the hostname of the host running the tests, such that the length is fixed?  Then using `{{ short_resource_prefix }}-codebuild-service-role`

Reply:
1) I agree. I never meant for that role to be used for anything other than sts_assume_role tests. A follow up PR would be good when someone has time for that. In addition to these, the ec2_instance tests need it removed.
2) Sure, maybe. There are a couple tests that do what I've done already so it would be convenient for those. Different AWS services have different lengths and shippable doesn't have a problem running the full length name, but locally I cannot as my computer name is too long.